src/components/Attachments/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,214 @@
+            const entry = _.first(viewableItems);
+            if (!entry) {
+                setActiveSource(null);
+                return;
+            }
+
+            setPage(entry.index);
+            setActiveSource(entry.item.source);
+
+            onNavigate(entry.item);
+        },
+        [onNavigate],
+    );
+
+    /**
+     * Increments or decrements the index to get another selected item
+     * @param {Number} deltaSlide
+     */
+    const cycleThroughAttachments = useCallback(
+        (deltaSlide) => {
+            const nextIndex = page + deltaSlide;
+            const nextItem = attachments[nextIndex];
+
+            if (!nextItem || !scrollRef.current) {
+                return;
+            }
+
+            scrollRef.current.scrollToIndex({index: nextIndex, animated: canUseTouchScreen});
+        },
+        [attachments, page],
+    );
+
+    /**
+     * Calculate items layout information to optimize scrolling performance
+     * @param {*} data
+     * @param {Number} index
+     * @returns {{offset: Number, length: Number, index: Number}}
+     */
+    const getItemLayout = useCallback(
+        (_data, index) => ({
+            length: containerWidth,
+            offset: containerWidth * index,
+            index,
+        }),
+        [containerWidth],
+    );
+
+    /**
+     * Defines how a container for a single attachment should be rendered
+     * @param {Object} cellRendererProps
+     * @returns {JSX.Element}
+     */
+    const renderCell = useCallback(
+        (cellProps) => {
+            // Use window width instead of layout width to address the issue in https://github.com/Expensify/App/issues/17760
+            // considering horizontal margin and border width in centered modal
+            const modalStyles = styles.centeredModalStyles(isSmallScreenWidth, true);
+            const style = [cellProps.style, styles.h100, {width: PixelRatio.roundToNearestPixel(windowWidth - (modalStyles.marginHorizontal + modalStyles.borderWidth) * 2)}];
+
+            return (
+                <View
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...cellProps}
+                    style={style}
+                />
+            );
+        },
+        [isSmallScreenWidth, windowWidth],
+    );
+
+    /**
+     * Defines how a single attachment should be rendered
+     * @param {Object} item
+     * @param {Boolean} item.isAuthTokenRequired
+     * @param {String} item.source
+     * @param {Object} item.file
+     * @param {String} item.file.name
+     * @returns {JSX.Element}
+     */
+    const renderItem = useCallback(
+        ({item}) => (
+            <AttachmentView
+                source={item.source}
+                file={item.file}
+                isAuthTokenRequired={item.isAuthTokenRequired}
+                isFocused={activeSource === item.source}
+                onPress={() => canUseTouchScreen && setShouldShowArrows(!shouldShowArrows)}
```

Comment:
I think this change introduced regression here https://github.com/Expensify/App/issues/25796